src/utilities/utility.ts

```diff
@@ -129,6 +137,75 @@ export function annotateChildPods(kubectl: Kubectl, resourceType: string, resour
     return commandExecutionResults;
 }
 
+export async function getFilePathsConfigs(): Promise<FileConfigPath> {
+
+    let filePathsConfig = <FileConfigPath>{};
+    const MANIFEST_PATHS_KEY = 'manifestFilePaths';
+    const HELM_CHART_KEY = 'helmChartFilePaths';
+    const DOCKERFILE_PATH_KEY = 'dockerfilePaths';
+    const DOCKERFILE_PATH_LABEL_KEY = 'dockerfile-path';
+
+    let inputManifestFiles = inputParams.manifests || [];
+    filePathsConfig[MANIFEST_PATHS_KEY] = inputManifestFiles;
+
+    let helmChartPaths = (process.env.HELM_CHART_PATHS && process.env.HELM_CHART_PATHS.split('\n').filter(path => path != "")) || [];
+    filePathsConfig[HELM_CHART_KEY] = helmChartPaths;
+
+    let imageNames = core.getInput('images').split('\n');
+    let imageDockerfilePathMap: any = {};
+    let pathValue: string, pathLink: string;
+    const branchOrTag: string = process.env.GITHUB_REF && process.env.GITHUB_REF.replace('refs/heads/','blob/').replace('refs/tags/','blob/');
+
+    //Fetching from image label if available
+    for (const image of imageNames) {
+        let args: string[] = [image];
+        let imageConfig: any;
+
+        try {
+            
+            await exec.exec('docker pull ', args, true).then(res => {
+                if (res.stderr != '' && !res.success) {
+                    throw new Error(`docker images pull failed with: ${res.stderr.match(/(.*)\s*$/)![0]}`);
+                }
+            });
+
+            await exec.exec('docker inspect --type=image', args, true).then(res => {
+                if (res.stderr != '' && !res.success) {
+                    throw new Error(`docker inspect call failed with: ${res.stderr.match(/(.*)\s*$/)![0]}`);
+                }
+
+                if (res.stdout) {
+                    imageConfig = JSON.parse(res.stdout);
+                }
+            });
+        }
+        catch (ex) {
+            core.warning(`Failed to get dockerfile paths for image ${image.toString()} | ` + ex);
+        }
+
+        if (imageConfig) {
+            imageConfig = imageConfig[0];
+            if ((imageConfig.Config) && (imageConfig.Config.Labels) && (imageConfig.Config.Labels[DOCKERFILE_PATH_LABEL_KEY])) {
+                pathValue = imageConfig.Config.Labels[DOCKERFILE_PATH_LABEL_KEY];
+                if(pathValue.startsWith('./')){  //if it is relative filepath convert to link from current repo
+                    pathLink = `https://github.com/${process.env.GITHUB_REPOSITORY}/${branchOrTag}/${pathValue}`;
+                    pathValue = pathLink;
+                }
+
+            }
+            else {
+                pathValue = 'Not available';
+            }
+                
+            imageDockerfilePathMap[image] = pathValue;
+        }
+    }
+
+    filePathsConfig[DOCKERFILE_PATH_KEY] = imageDockerfilePathMap;
+
+    return Promise.resolve(filePathsConfig);

```

Comment:
Just return `filePathsConfig`

Reply:
It is an async call that expects promise, since we are calling container registries via docker. It was throwing errors if I make it sync - why do we need to make it sync ?

Reply:
What error was it throwing? 

When you return `Promise<resolve>`, the expected js would be `Promise<Promise<resolve>>` right?

Reply:
No, it will return a Promise<FileConfigPath> object , which is what is expected from the await call to the function